src/libraries/System.Memory.Data/tests/BinaryDataTests.cs

```diff
@@ -614,6 +614,93 @@ public void ToStringReturnEmptyStringWhenBinaryDataEmpty()
             Assert.Equal(string.Empty, BinaryData.Empty.ToString());
         }
 
+        [Theory]
+        [InlineData(0)]
+        [InlineData(4)]
+        [InlineData(7)]
+        public void LengthReturnsNumberOfBytesForBinaryDataFromReadOnlyMemory(int count)
+        {
+            var data = BinaryData.FromBytes(new ReadOnlyMemory<byte>(new byte[count]));
+            Assert.Equal(count, data.Length);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(4)]
+        [InlineData(7)]
+        public void LengthReturnsNumberOfBytesForBinaryDataFromArray(int count)
+        {
+            var data = BinaryData.FromBytes(new byte[count]);
+            Assert.Equal(count, data.Length);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(4)]
+        [InlineData(7)]
+        public void LengthReturnsNumberOfBytesForBinaryDataFromString(int count)
+        {
+            var data = BinaryData.FromBytes(new byte[count]);
```

Comment:
the test name ends with `FromString` but you are using `FromBytes` here. I assume you meant using `FromString`?

```suggestion
            var data = BinaryData.FromString(new string('a', count));
```

Reply:
Absolutely! I just force-pushed 8f5328ae4ec.